rotkehlchen/tests/unit/test_data_updates.py

```diff
@@ -5,21 +5,71 @@
 
 from rotkehlchen.assets.asset import EvmToken
 from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.db.addressbook import DBAddressbook
 from rotkehlchen.db.updates import RotkiDataUpdater, UpdateType
 from rotkehlchen.errors.asset import UnknownAsset
 from rotkehlchen.globaldb.handler import GlobalDBHandler
 from rotkehlchen.tests.utils.factories import make_evm_address
 from rotkehlchen.tests.utils.mock import MockResponse
-from rotkehlchen.types import SPAM_PROTOCOL, ChainID, EvmTokenKind
+from rotkehlchen.types import (
+    SPAM_PROTOCOL,
+    AddressbookEntry,
+    ChainID,
+    EvmTokenKind,
+    SupportedBlockchain,
+)
 
 ETHEREUM_SPAM_ASSET_ADDRESS = make_evm_address()
 OPTIMISM_SPAM_ASSET_ADDRESS = make_evm_address()
 
+ADDRESSBOOK_ADDRESS_1 = make_evm_address()
+ADDRESSBOOK_ADDRESS_2 = make_evm_address()
+
+REMOTE_ADDRESSBOOK = [
+    AddressbookEntry(
+        name='Remote name that doesnt conflict with anything',
+        address=ADDRESSBOOK_ADDRESS_1,
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ), AddressbookEntry(
+        name='Remote name that has a conflict, and it should replace the local name',
+        address=ADDRESSBOOK_ADDRESS_2,
+        blockchain=SupportedBlockchain.OPTIMISM,
+    ), AddressbookEntry(
+        name='Remote name that has a conflict, but the local name should be preferred',
+        address=ADDRESSBOOK_ADDRESS_2,
+        blockchain=SupportedBlockchain.ETHEREUM,
+    ),
+]
+SERIALIZED_REMOTE_ADDRESSBOOK = [
+    {
+        'address': ADDRESSBOOK_ADDRESS_1,
+        'blockchain': 'ETH',
+        'new_name': 'Remote name that doesnt conflict with anything',
+        'old_name': None,
```

Comment:
I would advise putting `old_name` as optional. A lot less data to serialize for only an edge case. Only add `old_name` in the remote if it's needed. It's easy to guard against this in the app code as I showed you in the other comment and this really helps us as makes the size of the data in the remote and that needs to be serialized over the network smaller.